src/main/java/org/spongepowered/common/data/processor/value/entity/ElderValueProcessor.java

```diff
@@ -0,0 +1,83 @@
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.entity;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.monster.EntityGuardian;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+public class ElderValueProcessor extends AbstractSpongeValueProcessor<Boolean, Value<Boolean>> {
+
+    public ElderValueProcessor() {
+        super(Keys.ELDER_GUARDIAN);
+    }
+
+    @Override
+    protected Value<Boolean> constructValue(Boolean defaultValue) {
+        return new SpongeValue<Boolean>(Keys.ELDER_GUARDIAN, false, defaultValue);
+    }
+
+    @Override
+    public Optional<Boolean> getValueFromContainer(ValueContainer<?> container) {
+        if (container.supports(Keys.ELDER_GUARDIAN)) {
+            return Optional.of(((EntityGuardian) container).isElder());
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(ValueContainer<?> container) {
+        return container instanceof EntityGuardian;
+    }
+
+    @Override
+    public DataTransactionResult offerToStore(ValueContainer<?> container, Boolean value) {
+        final ImmutableValue proposedValue = new ImmutableSpongeValue(Keys.ELDER_GUARDIAN, value);
+        final ImmutableValue oldValue = getApiValueFromContainer(container).get().asImmutable();
```

Comment:
You're getting the previous value before you're checking whether the `container` is an `EntityGuardian`.